vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1233,6 +1234,8 @@ def _pool(
         hidden_states: torch.Tensor,
         num_scheduled_tokens: int,
         num_scheduled_tokens_np: np.ndarray,
+        finished_sending: Optional[set[str]],
```

Comment:
Why are all of these changes to `gpu_model_runner.py` and `gpu_worker.py` needed? I don't think they affect what's being fixed by this PR and so would be best to revert (unless I'm missing something).

Reply:
so ray wraps the runner directly, it's better for the model_runner to own the logic of filling in the finished_sending and finished_recving fields in the output. So I basically reverted the changes to gpu_model_runner introduced in https://github.com/vllm-project/vllm/pull/19555. In that PR, for some reason that I don't understand the logic is taken out of the model_runner and is put into the worker.

Reply:
@orozery Is there a test / script that I should target to make sure the intended pp logic still works with my changes?

Reply:
@orozery I see what you are saying. Basically for PP, each pp rank (except the last one) will forward an empty output back to the scheduler. You want those empty outputs to have the correct finished/recved attributes. I added that logic back but leaving the main model_runner logic intact. Let's get this merged asap so that it can catch the 0.10.0 train since it's a massive regression in the ray behavior. We can follow up with better solutions after.

Reply:
> not sure if there is a test for PP + P/D

Based on my research there is none. I also couldn't get the master run PP + P/D so not sure if at any point that combo was supported and now it's regressed? Or was it not supported at all.